### ISSUE TITLE
Support for Server Errors

### ISSUE THREAD

**Issue by GNSSYSTEMS-tg** [ghidoz/angular2-jsonapi#109]
I was playing around and found out that error handling is not correct according to the standard.

Server Response ([reference](http://jsonapi.org/format/upcoming/#error-objects)) with 422:
~~~
{
  "errors": [
    {
      "detail": "field may not be empty",
      "source": {
        "pointer": "/data/attributes/x"
      },
      "status":"422"
    }
  ]
}
~~~

Such a server response triggers [this](https://github.com/ghidoz/angular2-jsonapi/blob/ed59e60b0779ca1f659b4c0873faf5e44f8db997/src/services/json-api-datastore.service.ts#L96) line. First of all, I wonder why there is a `console.error` instead of an accurate logger. And second of all, why does this `catch` exists at all? Five lines below, there is another `catch` doing the right thing... Did I get something wrong?

**Comment by GNSSYSTEMS-tg** [comment]
Addition: removing the line found under the link above fixes the problem.

See PR #111

**Comment by HennerM** [comment]
Hey, that was not my intention of catching a possible error. I added the catch because of [this](https://github.com/ghidoz/angular2-jsonapi/blob/ed59e60b0779ca1f659b4c0873faf5e44f8db997/src/services/json-api-datastore.service.ts#L212) error that gets thrown if no body is present. I wanted to differentiate between server errors and errors occurred in parsing the response

**Comment by GNSSYSTEMS-tg** [comment]
Ok, I get the point. But the behaviour here is wrong. I suggest to split up both cases. Either the response is empty, thus a prepared model has to be emitted. Or there is an error and processing should be stopped.

**Comment by GNSSYSTEMS-tg** [comment]
Please have a look at PR #111 and add comments there if solution is insufficient. Otherwise merging is much appreciated.

**Comment by aimcom** [comment]
We had the same problem and fixed it by applying the changes from commit:98e1341. It would be nice if the PR #111 could be merged. :+1:

**Comment by HennerM** [comment]
Merged the PR, closing this.
